core/src/java/show/ezkz/casprzak/core/tools/functiongenerators/LegrendePolynomial.java

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * The methods in {@link LegrendePolynomial} deal with <a href="https://en.wikipedia.org/wiki/Legendre_polynomials">Legrende Polynomials</a>.
+*/
+public class LegrendePolynomial {
+
+	private static final Map<Integer, GeneralFunction> cache = new HashMap<>();
+
+	/**
+	 * Returns the nth Legrende polynomial with the default variable defined with {@link Settings#singleVariableDefault}
+	 * @param n the nth polynomial
+	 * @return nth Legrende polynomial
+	 */
+	public static GeneralFunction legrendePolynomial(int n) {
+		if (Settings.cacheDerivatives && cache.containsKey(n))
+			return cache.get(n);
+
+		int p = n % 2 == 0 ? n / 2 : (n + 1) / 2;
+		GeneralFunction[] sum = new GeneralFunction[n - p + 1];
+		for (int m = p; m <= n; m++) {
+			sum[m - p] = new Product(new Constant(constant(n, m)), new Pow(new Constant(2 * m - n), new Variable(Settings.singleVariableDefault))).simplify();
+		}
+		GeneralFunction polynomial = new Sum(sum).simplify();
+		cache.put(n, polynomial);
+		return polynomial;
+	}
+
+	/**
+	 * Returns the nth Legrende polynomial with a variable of the specified String
+	 * @param n the nth polynomial
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return nth Legrende polynomial
+	 */
+	public static GeneralFunction legrendePolynomial(int n, String variable) {
+		GeneralFunction polynomial = legrendePolynomial(n);
+		if (Settings.singleVariableDefault.equals(variable))
+			return polynomial;
+		else {
+			Map<String, Variable> substitution = new HashMap<>() {
+				{
+					put(Settings.singleVariableDefault, new Variable(variable));
+				}
+			};
+			return polynomial.substituteVariables(substitution);
+		}
+	}
+
+	/**
+	 * Returns the nth normalized Legrende polynomial with the default variable defined with {@link Settings#singleVariableDefault}
+	 * @param n the nth polynomial
+	 * @return nth normalized Legrende polynomial
+	 */
+	public static GeneralFunction normalLegrendePolynomial(int n) {
+		return new Product(new Constant(1 / normalizingConstantDouble(n)), legrendePolynomial(n)).simplify();
+	}
+
+	/**
+	 * Returns the nth normalized Legrende polynomial with a variable of the specified String
+	 * @param n the nth polynomial
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return nth normalized Legrende polynomial
+	 */
+	public static GeneralFunction normalLegrendePolynomial(int n, String variable) {
+		return new Product(new Constant(1 / normalizingConstantDouble(n)), legrendePolynomial(n, variable)).simplify();
+	}
+
+	/**
+	 * The normalizing constant of the nth Legrende polynomial
+	 * ie If you wanted to normalize, you would do {@code 1/N * L_n}
+	 * @param n the nth polynomial
+	 * @return normalizing constant of the nth Legrende polynomial
+	 */
+	public static GeneralFunction normalizingConstant(int n) {
+		return new Constant(normalizingConstantDouble(n));
+	}
+
+	private static double normalizingConstantDouble(int n) {
```

Comment:
Why does this method return a double while the others don't? Honestly having it return a `double` in all files is probably best (you can even make it public) anyways (that would also fix the unnecessary `reciprocal(Constant)` issue from before), but even if you don't do that still think about making this `1 / N`

Reply:
No?

Reply:
All the constant methods return doubles. The numbers aren't necessarily integers.

Reply:
What I meant is about returning a `double` or a `GeneralFunction`; some of them in the other files return `GeneralFunction`s and this returns `double`s. If you want the constants to be simplifiable and symbolic, you always want a `GF`, so this method shouldn't exist; if you don't care about that, then it's more efficient to just return a `double` in all those methods and make a `Constant` out of it